Ficha8/avl.c

```diff
@@ -0,0 +1,371 @@
+/*****************************************************************/
+/*          Arvore AVL | PROG2 | MIEEC | 2017/18                 */
+/*****************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "avl.h"
+ 
+/* declaracao das funcoes utilitarias */
+no_avl* avl_pesquisa_impl(no_avl *no, const char *str);
+no_avl* avl_insere_impl(no_avl *no, const char *str);
+no_avl* avl_remove_impl(no_avl *no, const char *str);
+no_avl* avl_no_valormin(no_avl* no);
+no_avl* avl_no_valormax(no_avl* no);
+void avl_preordem_impl(no_avl *no);
+
+
+arvore_avl* avl_nova()
+{
+    arvore_avl *avl = (arvore_avl*) malloc(sizeof(arvore_avl));
+    if(avl == NULL)
+        return NULL;
+
+    avl->raiz = NULL; /* arvore vazia */
+
+    return avl;
+}
+
+void avl_apaga(arvore_avl* avl)
+{
+	if (!avl) return;
```

Comment:
```suggestion
	if (avl == NULL) return;
```
The code assumed NULL is a 0 like value.